src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java

```diff
@@ -74,4 +70,54 @@ public User findUserByUsername(String username) {
 
         return userDtoService.createDto(entity);
     }
+
+    public void registerNewUser(UserRegisterForm newUser) {
+        // check username
+        String username = newUser.getUsername();
+
+        User user = null;
+        try {
+            user = this.findUserByUsername(newUser.getUsername());
+        } catch (UserNotFoundException ignored) {
+        }
+
+        if (null != user)
+            throw new UserAlreadyExistsException("Username already taken.");
+
+        // check pws.
+        String password = newUser.getPassword();
+        passwordIsValid(password);
+
+        String confirmationPassword = newUser.getConfirmationPassword();
+        passwordIsValid(confirmationPassword);
```

Comment:
not necessary. If the comparision is not succesful the error message will be more precise